src/building/new-target.md

```diff
@@ -0,0 +1,105 @@
+# Adding a new target
+
+These are a set of steps to add support for a new target. There are
+numerous end states and paths to get there, so not all sections may be
+relevant to your desired goal.
+
+## Specifying a new LLVM
+
+For very new targets, you may need to use a different fork of LLVM
+than what is currently shipped with Rust. In that case, navigate to
+the `src/llvm_project` git submodule, check out the appropriate commit
+for your fork, then commit that new submodule reference in the main
+Rust repository.
+
+An example would be:
+
+```
+cd src/llvm_project
+
+git remote add my-target-llvm some-llvm-repository
+
+git checkout my-target-llvm/my-branch
+
+cd ..
+
+git add llvm_target
+
+git commit -m 'Use my custom LLVM'
```

Comment:
```suggestion
cd src/llvm_project
git remote add my-target-llvm some-llvm-repository
git checkout my-target-llvm/my-branch
cd ..
git add llvm_target
git commit -m 'Use my custom LLVM'
```

Reply:
Sure. I find this harder to read, but if that's the convention.